docker/py/entrypoint.sh

```diff
@@ -1,37 +1,26 @@
 #!/bin/bash
 
-# Setup git user
-if [ -z "$(git config --global --get user.name)" ]; then
-    git config --global user.name "$GIT_AUTHOR_NAME"
-fi
-if [ -z "$(git config --global --get user.email)" ]; then
-    git config --global user.email "$EMAIL"
-fi
-
-# configure global git credentials
-if [[ -v CI_PROJECT ]];
+if [[ -v GIT_AUTHOR_NAME && -v EMAIL && -v CI_PROJECT ]];
 then
-    # set the global git credentials
-    git config --global credential.helper "store --file=/work/${CI_PROJECT}/.git/credentials"
-
-    # link to the home work directory
-    ln -sf /work/${CI_PROJECT} ~/work
+    # Setup git user
+    if [ -z "$(git config --global --get user.name)" ]; then
+        git config --global user.name "$GIT_AUTHOR_NAME"
+    fi
+    if [ -z "$(git config --global --get user.email)" ]; then
+        git config --global user.email "$EMAIL"
+    fi
+
+    # configure global git credentials
+    if [[ -v CI_PROJECT ]];
+    then
+        # set the global git credentials
+        git config --global credential.helper "store --file=/work/${CI_PROJECT}/.git/credentials"
+
+        # link to the home work directory
+        ln -sf /work/${CI_PROJECT} ~/work
+    fi
 fi
 
-#
-# copy the environment from renku-env repo
-#
-
-# clone the repo
-proto=$(echo $GITLAB_URL | sed -e's,^\(.*://\).*,\1,g')
-url=$(echo ${GITLAB_URL/$proto/})
-user=$(echo ${CI_REPOSITORY_URL/$proto/} | grep @ | cut -d@ -f1)
-
-git clone --depth 1 ${GITLAB_URL}/${JUPYTERHUB_USER}/renku-env.git /tmp/renku-env || true
-
-# append the contents of all the files to same files in ${HOME}
-find /tmp/renku-env -not -path '*.git*' -type f -print0 | xargs --null -I{} sh -c 'cat {} >> ${HOME}/$(basename "{}")' || true
-
 # install git hooks
 ~/.local/bin/renku githooks install || true
```

Comment:
For R image in ci-renku-2142, though the session got started, in the logs:

> Error: . is not a renku repository.
To initialize this as a renku repository use: renku init

Reply:
Good catch. This is actually a race condition which I've introduced by cloning the repo in the entrypoint of a "normal" container rather than an init container. I did this because I wanted to use the git-https proxy to manage the git connection _during_ the clone, however, this proxy is no available during the execution of init containers. I've now removed that race condition by adding a small dummy file to the repo when the clone is done, and by waiting for that dummy file to appear inside the entrypoint of the jupyter server container.

Reply:
thanks @ableuler, it's working now 👍